src/uu/sort/src/sort.rs

```diff
@@ -266,15 +264,17 @@ type Field = Range<usize>;
 
 #[derive(Clone)]
 pub struct Line {
-    line: String,
+    line: Box<str>,
     // The common case is not to specify fields. Let's make this fast.
-    selections: SmallVec<[Selection; 1]>,
+    first_selection: Selection,
+    other_selections: Box<[Selection]>,
```

Comment:
Getting rid of smallvec is not a memory usage improvement, but an overall performance improvement.

Reply:
To be more precise: This performance improvement is noticeable when a custom key (-k) is in use. For such cases this is reducing memory by about 30% (`/bin/time -v target/release/coreutils sort shuffled_wordlist.txt -k 2.1 > /dev/null`) and improves speed by 20% (`hyperfine "target/release/coreutils sort shuffled_wordlist.txt -k 2.2" "./oldutils sort shuffled_wordlist.txt -k 2.2"`)